### DISCUSSION TITLE
Python Webdriver Selenium HELP !

### DISCUSSION THREAD

**Discussion by qusai2316** [community/community#21364] [Other Feature Feedback, Questions, & Ideas]
<p>Im using Python Webdriver Selenium for web scraping i know it is not the best way to scraping , anyway when i used ID to get page text some blocks that includ text did not copy its seems to locked i want to unlock it , im not interface with stable page that can you edit HTML manualy im using code to put some digit that give me a result which copy it then reply this prosses with new digit after copyed the last text result with “while” and so on .</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/0/0c3edca6d0fa1dc78765bb0339c1b2572db35a3b.jpeg" data-download-href="/uploads/short-url/1KkpsGCKgmgCcBLUigHEwGyR18f.jpg?dl=1" title="ioot.jpg" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/28514750/181063878-b23e619b-11f1-4069-ae08-8c2e150f8b36.jpg" alt="ioot.jpg" data-base62-sha1="1KkpsGCKgmgCcBLUigHEwGyR18f" width="690" height="303" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/0/0c3edca6d0fa1dc78765bb0339c1b2572db35a3b_2_690x303.jpg, https://aws1.discourse-cdn.com/github/optimized/2X/0/0c3edca6d0fa1dc78765bb0339c1b2572db35a3b_2_1035x454.jpg 1.5x, https://aws1.discourse-cdn.com/github/original/2X/0/0c3edca6d0fa1dc78765bb0339c1b2572db35a3b.jpeg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/0/0c3edca6d0fa1dc78765bb0339c1b2572db35a3b_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">ioot.jpg</span><span class="informations">1086×478 111 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>This is the code </p>
<p>username=2000234361</p>
<p>while (username&lt;2000234371):<br>
driver.find_element_by_id(‘txtNatID’).send_keys(username)<br>
driver.find_element_by_id(‘btnSearch’).click()<br>
name=driver.find_element_by_id(“txtNameAR”).text<br>
name_en=driver.find_element_by_id(“txtNameEN”).text<br>
username+=1<br>
print(username-1,"*",name,"*",name_en)<br>
driver.find_element_by_id(“txtNatID”).clear()</p>

**Comment by fire-eggs** [comment]
<p>I’m having difficulty following what you are trying to accomplish and what the problem(s) are.</p>
<p>I _think_ the first problem you say you are having is this line:</p>
<pre><code>name_en=driver.find_element_by_id("txtNameEN").text
</code></pre>
<p>does not work because the “txtNameEN” field is disabled. Some googling suggests you might try the “value” property, such as:</p>
<pre><code>name_en=driver.find_element_by_id("txtNameEN").value
</code></pre>

**Comment by qusai2316** [comment]
<p>Unfortunately it didn’t work</p>
<p>Traceback (most recent call last):<br>
File “&lt;stdin&gt;”, line 4, in &lt;module&gt;<br>
AttributeError: ‘WebElement’ object has no attribute ‘value’</p>

**Comment by fire-eggs** [comment]
<p>OK, try:</p>
<pre><code>driver.findElement(By.id("write_element_id_here")).getAttribute("value");
</code></pre>
<p>substituting “write_element_id_here” appropriately.</p>
<p>Not a Selenium expert, merely googling “selenium disabled text field”.</p>

**Comment by qusai2316** [comment]
<p>Also it did not work :</p>
<p>AttributeError: ‘WebElement’ object has no attribute ‘getAttribute’</p>
<p>Most of Selenium tutorials on java nor python</p>

**Comment by fire-eggs** [comment, marked as answer]
<p>Whelp, my Selenium ignorance is showing.</p>
<p><a href="https://stackoverflow.com/questions/46711144/how-to-get-text-from-disabled-field-in-selenium-using-python" rel="nofollow noopener">This</a> suggests it is “.get_attribute(‘value’)”, not “.getAttribute(“value”)”.  Which is confirmed by the <a href="https://seleniumhq.github.io/selenium/docs/api/py/webdriver_remote/selenium.webdriver.remote.webelement.html?highlight=webelement#selenium.webdriver.remote.webelement" rel="nofollow noopener">docs</a>.</p>

**Comment by qusai2316** [comment]
<p>Finally it is work !!</p>
<p>Thank you</p>
